LanguageFeatures/Records/constants_A02_t02.dart

```diff
@@ -0,0 +1,34 @@
+// Copyright (c) 2022, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion A record expression is a potentially constant expression if and
+/// only if all its field expressions are potentially constant or constant
+/// expressions. This means that a record expression can be used in the
+/// initializer list of a constant non-redirecting generative constructor, and
+/// can depend on constructor parameters.
+///
+/// @description Checks that a record expression is not a potentially constant
+/// expression if any of its field expressions is not a potentially constant
+/// or constant expressions
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=records
+
+final Pi = 3.14;
+
+class C {
+  final (num, List<int>?) r;
+  const C.c1(int i): r = (i + 1, const [42]);
```

Comment:
Sorry for a typo here. Intention was 

```dart
   const C.c1(int i): r = (i + 1, const [i]);
```
This, I believe, is not a potentially constant

Reply:
Right, but this is still not testing anything about records: It just tests that `const [i]` is a compile-time error when `i` is not a constant expression, and it makes no difference that this error occurs inside a record literal, or inside any other expression. `Pi` is a safer bet than `const [...]`, because the latter is either an error in itself, or it is a constant expression (hence also potentially constant).

Reply:
Changed to
```dart
const C.c1(int i): r = (i + 1, Pi);
```